mysql-test/suite/clone/t/local_basic_gtid_compression.test

```diff
@@ -0,0 +1,122 @@
+# 3. Install Clone Plugin, enable debug point to print compression info
+#    and initiate Clone operation.
+# 4. Assert that table is compressed after Clone operation.
+# 5. Assert that the compression was done by the gtid persister thread.
+# 6. Cleanup
+#
+# ==== References ====
+#
+# PS-6990: Gaps in mysql.gtid_executed but not in @@GLOBAL.gtid_executed
+
+# This test requires debug binaries
+--source include/have_debug.inc
+# This test requires binlogging.
+--source include/have_log_bin.inc
+
+--let $server_uuid = `SELECT @@GLOBAL.SERVER_UUID`
+
+# Clean the table before starting the test.
+RESET MASTER;
+
+# Test setup for clone test.
+--source ../include/clone_connection_begin.inc
+--let $CLONE_DATADIR = $MYSQL_TMP_DIR/data_new
+
+--echo #
+--echo # 1. Create InnoDB and MyISAM tables and insert 2 rows.
+CREATE TABLE t1(col1 INT PRIMARY KEY, col2 char(64)) ENGINE = INNODB;
+CREATE TABLE t2(col1 INT PRIMARY KEY, col2 char(64)) ENGINE = MyISAM;
+
+INSERT INTO t2 VALUES(10, 'myisam row 1');
+INSERT INTO t1 VALUES(10, 'innodb row 1');
+INSERT INTO t2 VALUES(20, 'myisam row 2');
+INSERT INTO t1 VALUES(20, 'innodb row 2');
+
+--echo #
+--echo # 2. Assert that mysql.gtid_executed table is not compressed and has gaps.
+
+# Assert that 6 transactions have been committed in total.
+--let $assert_text = COMMITTED_GTIDS SERVER_UUID:1-6
+--let $assert_cond = "[SELECT @@GLOBAL.GTID_EXECUTED]" = "$server_uuid:1-6"
+--source include/assert.inc
+
+# When binlog is enabled, GTIDs of InnoDB transactions are updated by gtid
+# persister thread and GTIDs of other transactions are updated on binlog
+# rotation. So, wait till the GTIDs of InnoDB transactions are updated by the
+# gtid persister thread.
+#
+# i.e, wait till the below state is reached.
+#
+# SELECT * FROM mysql.gtid_executed;
+# source_uuid	interval_start	interval_end
+# ------------------------------------------
+# server_uuid	             1	           1
+# server_uuid	             4	           4
+# server_uuid	             6	           6
+
+--let $wait_condition = SELECT count(*) = 3 FROM mysql.gtid_executed
+--source include/wait_condition.inc
+
+--let $assert_cond = COUNT(*) = 3 FROM mysql.gtid_executed
+--let $assert_text = The mysql gtid table is not compressed and has gaps.
+--source include/assert.inc
+
+--echo #
+--echo # 3. Install Clone Plugin, enable debug point to print compression info
+--echo #    and initiate Clone operation.
+
+# Install Clone Plugin
+--replace_result $CLONE_PLUGIN CLONE_PLUGIN
+--eval INSTALL PLUGIN clone SONAME '$CLONE_PLUGIN'
+
+# Enable debug point to print the compression info for validation purpose.
+--let $debug_point = print_gtid_compression_info
+--source include/add_debug_point.inc
+
+# Clone data
+--connection clone_conn_1
+--source ../include/clone_command.inc
+
+# During clone operation, persister thread, along with writing InnoDB GTIDs, it
+# also writes non-InnoDB GTIDs and even compresses it by creating an explicit
+# request.
```

Comment:
How will gtid persistor thread know about the gtids commited outside InnoDB?

Reply:
* When clone is [synchronising](https://github.com/percona/percona-server/blob/8.0/storage/innobase/clone/clone0copy.cc#L269) with the binary log, it creates an explicit request for a flush and enters wait. 
* When persister thread receives an explicit request, it updates [non-innodb gtids](https://github.com/percona/percona-server/blob/8.0/storage/innobase/clone/clone0repl.cc#L473) and also compresses it.

Reply:
ah.. that way.. but the gtids are are not inserted into mysql.gtid_executed table by InnoDB or persistor thread right? Who will insert into mysql.gtid_executed table for non InnoDB transactions?

Reply:
Persister thread itself writes that.
```
  307 int Clone_persist_gtid::write_other_gtids() {
  308   int err = 0;
  309   if (opt_bin_log) {
  310     err = gtid_state->save_gtids_of_last_binlog_into_table();
  311   }
  312   return (err);
  313 }
```

Reply:
Thanks! its clear now